packages/flutter/test/material/time_picker_test.dart

```diff
@@ -222,6 +222,48 @@ void main() {
     expect(selectedLabels.map<bool>((dynamic tp) => tp.inner as bool), inner0To23);
   });
 
+  // Regression test for https://github.com/flutter/flutter/issues/164860
+  testWidgets('Material3 - formats 24-hour numbers correctly in Farsi', (
+    WidgetTester tester,
+  ) async {
+    await mediaQueryBoilerplate(
+      tester,
+      locale: const Locale('fa', 'IR'),
+      materialType: MaterialType.material3,
+    );
+
+    final MaterialLocalizations localization = MaterialLocalizations.of(
+      tester.element(find.byType(TimePickerDialog)),
+    );
+
+    final List<String> labels00To23 = List<String>.generate(24, (int index) {
+      return localization.formatHour(
+        TimeOfDay(hour: index, minute: 0),
+        alwaysUse24HourFormat: true,
+      );
+    });
+    final List<bool> inner0To23 = List<bool>.generate(24, (int index) => index >= 12);
+
+    final CustomPaint dialPaint = tester.widget(findDialPaint);
+    final dynamic dialPainter = dialPaint.painter;
+    // ignore: avoid_dynamic_calls
```

Comment:
Why are we using dynamic through the test?

Reply:
@Piinks since dialPainter is an instance of `_DialPainter` and which is private. That's why to access primaryLabels,  dynamic type was used.

Reply:
Oh. I see we're doing this in other tests. Unfortunate, but I see why. Maybe in the future we should consider cleaning these up and maybe making the private class visible for testing.